go/cli/mcap/cmd/cat.go

```diff
@@ -109,57 +109,68 @@ func printMessages(
 			}
 			continue
 		}
-		switch schema.Encoding {
-		case "ros1msg":
-			transcoder, ok := transcoders[channel.SchemaID]
-			if !ok {
-				packageName := strings.Split(schema.Name, "/")[0]
-				transcoder, err = ros.NewJSONTranscoder(packageName, schema.Data)
-				if err != nil {
-					return fmt.Errorf("failed to build transcoder for %s: %w", channel.Topic, err)
+		if schema == nil || schema.Encoding == "" {
+			switch channel.MessageEncoding {
+			case "json":
+				if _, err = msg.Write(message.Data); err != nil {
+					return fmt.Errorf("failed to write message bytes: %w", err)
 				}
-				transcoders[channel.SchemaID] = transcoder
-			}
-			msgReader.Reset(message.Data)
-			err = transcoder.Transcode(msg, msgReader)
-			if err != nil {
-				return fmt.Errorf("failed to transcode %s record on %s: %w", schema.Name, channel.Topic, err)
+			default:
+				return fmt.Errorf("For schema-less channels, JSON output is only supported with 'json' message encoding. Found: %s", channel.MessageEncoding)
```

Comment:
Should we use `self-describing` as the terminology for `schema-less` to be consistent? @james-rms

Reply:
Seems that both terminology is used:
https://github.com/foxglove/mcap/search?q=self-describing
https://github.com/foxglove/mcap/search?q=schemaless